srcStatic/GameModules/DllModule.cpp

```diff
@@ -18,4 +19,56 @@ void DllModule::LoadModuleDll(const std::string& dllPath)
 	}
 
 	this->moduleDllHandle = dllHandle;
+
+	DetectExportedModuleFunctions();
+}
+
+void DllModule::DetectExportedModuleFunctions()
+{
+	loadModuleFunctionIni = (LoadModuleFunctionIni)GetProcAddress(moduleDllHandle, "InitMod");
+	if (loadModuleFunctionIni == nullptr) {
+		loadModuleFunctionConsole = (LoadModuleFunctionConsole)GetProcAddress(moduleDllHandle, "mod_init");
+	}
+
+	unloadModuleFunction = (UnloadModuleFunction)GetProcAddress(moduleDllHandle, "DestroyMod");
+	if (unloadModuleFunction == nullptr) {
+		unloadModuleFunction = (UnloadModuleFunction)GetProcAddress(moduleDllHandle, "mod_destroy");
+	}
+
+	runModuleFunction = (RunModuleFunction)GetProcAddress(moduleDllHandle, "RunMod");
+	if (runModuleFunction == nullptr) {
+		runModuleFunction = (RunModuleFunction)GetProcAddress(moduleDllHandle, "mod_run");
+	}
+}
+
+void DllModule::Load()
+{
+	if (loadModuleFunctionIni != nullptr) {
+		loadModuleFunctionIni(Name().c_str());
+	}
+	else if (loadModuleFunctionConsole != nullptr) {
+		loadModuleFunctionConsole();
+	}
+}
+
+bool DllModule::Unload()
+{
+	bool success = true;
+
+	if (unloadModuleFunction != nullptr) {
+		success = unloadModuleFunction();
+	}
+
+	if (moduleDllHandle != nullptr) {
+		FreeLibrary(moduleDllHandle);
+	}
+
+	return success;
```

Comment:
This return value is ignored. Perhaps we should consider getting rid of it. The best we can really do with a module that fails to unload is to log the error.

Reply:
good point about the difference between the two unload functions.

What do you think if we just change the specification for a console module to have a void return? Then they would both read:

```C++
typedef void(*UnloadModuleFunction)();
```

I just tested, and it looks like the cm1 module loads fine as a console module with the code tweaked. I could just pull out the success checks altogether.

Reply:
It is tempting. I'd thought of doing that before. It should be safe to change the return type from `bool` to `void`.

I noticed the NetHelper module can actually return a failure result there.

I think I have a slight preference for retaining the official export signature as it was, and then just swallowing the result value in the `DllModule::Unload` method, with a possible log message in case of failure.

Ideally, modules should do their own error reporting, as it would be more detailed, however, older existing modules wouldn't have really had that option.

Reply:
I spoke with Arklon about this. It seems the return value was added as a way to potentially support a future in-game mod manager that could handle hot loading and unloading of modules, without a game restart. Potentially some mods would be tied to specific missions, such as the survivor missions.

Reply:
OK, I'm good with adding both signatures. I'll have it check for the bool version first, and if it isn't present, pull the void return version.